### ISSUE TITLE
add score method to GAMs

### ISSUE THREAD

**Issue by dswah** [dswah/pyGAM#102]
could be nice to have a built-in `gam.score` method like sklearn does on its estimators...

**Comment by edwardcjohnson** [comment]
If the feature functions f_i() can be returned as well as the coefficients from the fitted model, then this score method would simply be the product of those two after sending your new X through f_i().

**Comment by dswah** [comment]
more precisely im looking for a method that scores predictions against a labeled dataset. this should similar to the [sklearn functionality](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.score) where regression models return R-squared, classification modela return accuracy

the reason is for better out-of-the-box compatibility with sklearn gridsearchcv and randomizedsearchcv.
